src/ln/msgs.rs

```diff
@@ -782,6 +782,12 @@ impl From<::std::io::Error> for DecodeError {
 	}
 }
 
+impl From<::bitcoin_hashes::Error> for DecodeError {
```

Comment:
Hmm, I think you can skip this impl and just unwrap/expect in the impl Readable for Sha256dHash - the only Err return value is when you dont provide the right length value but we always do there.